JitPack build fails

```
Build starting...
Start: Sat Feb 25 03:09:23 UTC 2023 0e50a0ce3637
Git:
v6.2.0-0-g67e75f3
commit 67e75f30d569552f56a817ffcb645a0ad5d7901e
Author: Tobias Specht 
Date:   Fri Feb 24 17:05:05 2023 +0100

    Merge all include paths from compilation database for unity build mode (#1107)
    

Init SDKMan
Running before_install command:
sdk install java 17.0.5-oracle;sdk use java 17.0.5-oracle;mkdir -p ~/go;wget -q https://go.dev/dl/go1.18.3.linux-amd64.tar.gz && tar -C ~ -xzf go1.18.3.linux-amd64.tar.gz;ls -l ~/go
[1;33mWe periodically need to update the local cache. Please run:[0m

  $ sdk update

[1;36mBroadcasts discontinued. Please follow @sdkman_ on Twitter for announcements.[0m

Downloading: java 17.0.5-oracle

In progress...


                                                                           0.6%
##                                                                         4.1%
######                                                                     9.0%
#########                                                                 13.4%
############                                                              17.5%
################                                                          23.1%
###################                                                       27.8%
########################                                                  33.9%
###########################                                               38.5%
###############################                                           44.0%
##################################                                        47.4%
###################################                                       49.6%
#######################################                                   54.6%
##########################################                                58.9%
#############################################                             62.7%
##############################################                            64.9%
#################################################                         68.7%
#####################################################                     74.6%
#########################################################                 79.4%
###########################################################               83.1%
###############################################################           88.5%
###################################################################       94.3%
####################################################################      95.2%
#####################################################################     96.4%
######################################################################## 100.0%

Repackaging Java 17.0.5-oracle...

Done repackaging...

[1;32mInstalling: java 17.0.5-oracle[0m
[1;32mDone installing![0m


[1;32mSetting java 17.0.5-oracle as default.[0m
[1;33mWe periodically need to update the local cache. Please run:[0m

  $ sdk update


[1;32mUsing java version 17.0.5-oracle in this shell.[0m
total 236
-rw-r--r--  1 jitpack jitpack  56057 Jun  1  2022 AUTHORS
-rw-r--r--  1 jitpack jitpack   1339 Jun  1  2022 CONTRIBUTING.md
-rw-r--r--  1 jitpack jitpack 111408 Jun  1  2022 CONTRIBUTORS
-rw-r--r--  1 jitpack jitpack   1479 Jun  1  2022 LICENSE
-rw-r--r--  1 jitpack jitpack   1303 Jun  1  2022 PATENTS
-rw-r--r--  1 jitpack jitpack   1475 Jun  1  2022 README.md
-rw-r--r--  1 jitpack jitpack    397 Jun  1  2022 SECURITY.md
-rw-r--r--  1 jitpack jitpack      8 Jun  1  2022 VERSION
drwxr-xr-x  2 jitpack jitpack   4096 Jun  1  2022 api
drwxr-xr-x  2 jitpack jitpack   4096 Jun  1  2022 bin
-rw-r--r--  1 jitpack jitpack     52 Jun  1  2022 codereview.cfg
drwxr-xr-x  2 jitpack jitpack   4096 Jun  1  2022 doc
drwxr-xr-x  3 jitpack jitpack   4096 Jun  1  2022 lib
drwxr-xr-x 12 jitpack jitpack   4096 Jun  1  2022 misc
drwxr-xr-x  6 jitpack jitpack   4096 Jun  1  2022 pkg
drwxr-xr-x 48 jitpack jitpack   4096 Jun  1  2022 src
drwxr-xr-x 27 jitpack jitpack  12288 Jun  1  2022 test
Running install command:
export PATH="$PATH:$HOME/go/bin";./gradlew build -xtest -Pgroup=com.github.Fraunhofer-AISEC -PnodeDownload=true publishToMavenLocal
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8 -Dhttps.protocols=TLSv1.2
Downloading https://services.gradle.org/distributions/gradle-8.0.1-bin.zip
...........10%............20%............30%............40%............50%............60%...........70%............80%............90%............100%

Welcome to Gradle 8.0.1!

Here are the highlights of this release:
 - Improvements to the Kotlin DSL
 - Fine-grained parallelism from the first build with configuration cache
 - Configurable Gradle user home cache cleanup

For more details see https://docs.gradle.org/8.0.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Settings file '/home/jitpack/build/settings.gradle.kts' line: 24

* What went wrong:
Cannot get non-null property 'enableGoFrontend' on settings 'cpg' as it does not exist

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 27s
Build tool exit code: 0
Looking for artifacts...
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8 -Dhttps.protocols=TLSv1.2
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8 -Dhttps.protocols=TLSv1.2
Looking for pom.xml in build directory and ~/.m2
2023-02-25T03:10:23.148328638Z
Exit code: 0

⚠️ ERROR: No build artifacts found
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -65,32 +65,32 @@ sonarqube {
 // Load the properties that define which frontends to include
 //
 // this code block also exists in settings.gradle.kts
-val enableJavaFrontend by extra {
-    val enableJavaFrontend: String by project
+val enableJavaFrontend: Boolean by extra {
+    val enableJavaFrontend: String? by project
     enableJavaFrontend.toBoolean()
 }
 project.logger.lifecycle("Java frontend is ${if (enableJavaFrontend) "enabled" else "disabled"}")
 
-val enableGoFrontend by extra {
-    val enableGoFrontend: String by project
+val enableGoFrontend: Boolean by extra {
+    val enableGoFrontend: String? by project
     enableGoFrontend.toBoolean()
 }
 project.logger.lifecycle("Go frontend is ${if (enableGoFrontend) "enabled" else "disabled"}")
 
-val enablePythonFrontend by extra {
-    val enablePythonFrontend: String by project
+val enablePythonFrontend: Boolean by extra {
+    val enablePythonFrontend: String? by project
     enablePythonFrontend.toBoolean()
 }
 project.logger.lifecycle("Python frontend is ${if (enablePythonFrontend) "enabled" else "disabled"}")
 
-val enableLLVMFrontend by extra {
-    val enableLLVMFrontend: String by project
+val enableLLVMFrontend: Boolean by extra {
+    val enableLLVMFrontend: String? by project
     enableLLVMFrontend.toBoolean()
 }
 project.logger.lifecycle("LLVM frontend is ${if (enableLLVMFrontend) "enabled" else "disabled"}")
 
-val enableTypeScriptFrontend by extra {
-    val enableTypeScriptFrontend: String by project
+val enableTypeScriptFrontend: Boolean by extra {
+    val enableTypeScriptFrontend: String? by project
     enableTypeScriptFrontend.toBoolean()
 }
 project.logger.lifecycle("TypeScript frontend is ${if (enableTypeScriptFrontend) "enabled" else "disabled"}")

--- a/jitpack.yml
+++ b/jitpack.yml
@@ -6,4 +6,5 @@ before_install:
   - ls -l ~/go
 install:
   - export PATH="$PATH:$HOME/go/bin"
-  - ./gradlew build -xtest -Pgroup=com.github.Fraunhofer-AISEC -PnodeDownload=true publishToMavenLocal
+  - cp gradle.properties.example gradle.properties
+  - ./gradlew build -xtest -Pgroup=com.github.Fraunhofer-AISEC -PnodeDownload=true -PenableJavaFrontend=true -PenableGoFrontend=true -PenablePythonFrontend=true -PenableLLVMFrontend=true -PenableTypeScriptFrontend=true publishToMavenLocal

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -20,24 +20,24 @@ include(":cpg-neo4j")
 include(":cpg-console")
 
 // this code block also exists in the root build.gradle.kts
-val enableJavaFrontend by extra {
-    val enableJavaFrontend: String by settings
+val enableJavaFrontend: Boolean by extra {
+    val enableJavaFrontend: String? by settings
     enableJavaFrontend.toBoolean()
 }
-val enableGoFrontend by extra {
-    val enableGoFrontend: String by settings
+val enableGoFrontend: Boolean by extra {
+    val enableGoFrontend: String? by settings
     enableGoFrontend.toBoolean()
 }
-val enablePythonFrontend by extra {
-    val enablePythonFrontend: String by settings
+val enablePythonFrontend: Boolean by extra {
+    val enablePythonFrontend: String? by settings
     enablePythonFrontend.toBoolean()
 }
-val enableLLVMFrontend by extra {
-    val enableLLVMFrontend: String by settings
+val enableLLVMFrontend: Boolean by extra {
+    val enableLLVMFrontend: String? by settings
     enableLLVMFrontend.toBoolean()
 }
-val enableTypeScriptFrontend by extra {
-    val enableTypeScriptFrontend: String by settings
+val enableTypeScriptFrontend: Boolean by extra {
+    val enableTypeScriptFrontend: String? by settings
     enableTypeScriptFrontend.toBoolean()
 }
 

